### DISCUSSION TITLE
Varia UT800 + Giant HL1800 support needed

### DISCUSSION THREAD

**Discussion by EyessOnlyy** [maca88/SmartBikeLights#52] [General]
Hello Maca, 

First of all - **big thank you for your datafield**. Using it since a long time and very happy with it. 
I was using it with Varia UT800 + RTL515 radar -> all OK

I recently bought a Giant HL1800 and trying to connect both headlights in the same time -> from your documentation it seems it should be possible (by displaying only the modes suppprted by both lamps)
But, I get Error 3 when both headlights are connected to my Edge1040. 

UT800 + radar -> OK
HL1800 + radar -> OK
UT800 + HL1800 + radar -> NOK Error 3

Can you help me identifying what's wrong?

I have 3 configs. Selecting any of them when all ligths are connected to Edge1040 does not work

------------------------------
Head&Tail

1,1!Record:1H=3#0,73413136::1#5,4!High:2:2:0:0C]8.33333Es0,r0!Night:1:3:0:0Es0,r0!Day:1:7:0:0Er0,s0!:1:7:0:0D=1#0,73535488::1#5,4!BRAKE:1:4:0:0A[-10!DAY:2:7:0:0Er0,s0I[400]0!NIGHT:1:6:0:0Es0,r0!:1:0:0:0D=1#7,6:Varia 800:0:16777215!2,:-1,Off:0!1,High:2!1,Medium:3!1,Low:4!1,Night Flash:6!1,Day Flash:7#6,5:EyesOnly's Radar:0:16777215!1,:-1!1,Day Flash:7!1,Peloton:5!1,Night Flash:6!2,Solid:4,Off:0#0::#0:0#123!:123!#0#B3843#4#3#0#0
-----------------------------

-----------------------------
Giant HL1800:

1,1!Record:1H=3#0,1073754640::1#5,4!High:2:2:0:0C]8.33333Es0,r0!Night:1:3:0:0Es0,r0!Day:1:8:0:0Er0,s0!:1:8:0:0D=1#0,73535488::1#5,4!BRAKE:1:4:0:0A[-10!DAY:2:7:0:0Er0,s0I[400]0!NIGHT:1:6:0:0Es0,r0!:1:0:0:0D=1#6,5:HL 1800:0:16777215!2,:-1,Off:0!1,High:2!1,Medium:3!1,Low:4!1,Day Flash:8#6,5:EyesOnly's Radar:0:16777215!1,:-1!1,Day Flash:7!1,Peloton:5!1,Night Flash:6!2,Solid:4,Off:0#0::#0:0#123!:123!#0#0#B3843#7#3#0#0
-----------------------------

-----------------------------
Test2lights no night flash Garmin:

1,1!Record:1H=3#0,73413136::1#5,4!High:2:2:0:0C]8.33333Es0,r0!Night:1:3:0:0Es0,r0!Day:1:7:0:0Er0,s0!:1:7:0:0D=1#0,73535488::1#5,4!BRAKE:1:4:0:0A[-10!DAY:2:7:0:0Er0,s0I[400]0!NIGHT:1:6:0:0Es0,r0!:1:0:0:0D=1#6,5:Varia 800:0:16777215!2,:-1,Off:0!1,High:2!1,Medium:3!1,Low:4!1,Day Flash:7#6,5:EyesOnly's Radar:0:16777215!1,:-1!1,Day Flash:7!1,Peloton:5!1,Night Flash:6!2,Solid:4,Off:0#0::#0:0#123!:123!#0#0#B3843#4#3#0#0
-----------------------------

**Comment by maca88** [comment]
Hi,

In order to solve the issue you need to set the serial number for Giant HL1800 and UT800 headlights in all three configurations. Without setting the serial number, Smart Bike Lights will combine both headlights into one virtual light which you want to avoid.

**Reply by EyessOnlyy** [reply]
Thank you! I will try that

I was, wrongly,  under impression that serial number is needed when two lights of the same brand are connected not two lights of the same purpose type
I know read again the help info and read correctly, this time :D

![image](https://github.com/maca88/SmartBikeLights/assets/152268043/af5ce92f-2e1f-4e24-912c-1db041833d6e)

**Reply by maca88** [reply]
> I recently bought a Giant HL1800 and trying to connect both headlights in the same time

My bad, the serial number is required only when you want to control one of the headlights. If you want to control both of them at the same time, then I would suggest to also install `Smart Bike Lights #2` and configure one of the headlights on the second data field. Do note that the serial number is still required for all configurations (headlights only) in order to avoid Error 3. 

If you really want to use only one data field, then you need to create a custom configuration that will include only the light modes that both Giant HL1800 and UT800 by comparing the light mode numbers. 

Giant HL1800 light modes:
2 -> High
3 -> Medium
4 -> Low
8 -> Day Flash  

UT800 light modes:
2 -> High
3 -> Medium
4 -> Low
6 -> Night Flash
7 -> Day Flash

Based on the above light modes, only three modes do match:
2 -> High
3 -> Medium
4 -> Low

so you can create a custom configuration only for those, which means that you won't be able to use Day Flash/Night Flash modes.

**Comment by EyessOnlyy** [comment]
Update - tried to add the serial numbers but it seems that for Giant HL1800 I have no sensor information - all fields(sw version, serial number....) are listed with value 0 :(

I've also tried with the Giant connected to my Fenix7 watch - same 0 numbers in the sensor details

**Reply by maca88** [reply]
You are right, I tested with my Giant HL1800 and I also see all fields with 0 value, very strange. Try to use 0 as the serial number in the Lights Configurator and let me know if it works.

**Reply by EyessOnlyy** [reply]
i've tried that. it seems it does not consider 0. I add the 0 and then copy the config - it is unchanged in content
When I load it to check it the details - the 0 is gone

**Reply by maca88** [reply]
I've modified the Lights Configurator to support zero as a serial number. I did a test with Giant HL1800 and it is working for me.

**Reply by EyessOnlyy** [reply]
I've just tested with zero serial number Works like a charm
- switching configs in one data field
- using 2nd data field to have them both controlled 

THANK YOU!
